### PR TITLE
.github/worflows: add go vulnerability management check for cri-o

### PR DESCRIPTION
**Pull request by sohankunkerkar**
#### What type of PR is this?

/kind ci

#### Does this PR introduce a user-facing change?

```release-note
None
```

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -12,6 +12,7 @@
 [![FOSSA Status](https://app.fossa.io/api/projects/git%2Bgithub.com%2Fcri-o%2Fcri-o.svg?type=shield)](https://app.fossa.io/projects/git%2Bgithub.com%2Fcri-o%2Fcri-o?ref=badge_shield)
 [![Mentioned in Awesome CRI-O](https://awesome.re/mentioned-badge.svg)](awesome.md)
 [![Gitpod ready-to-code](https://img.shields.io/badge/Gitpod-ready--to--code-blue?logo=gitpod)](https://gitpod.io/#https://github.com/cri-o/cri-o)
+[![govulncheck](https://github.com/cri-o/cri-o/actions/workflows/govulncheck.yml/badge.svg)](https://github.com/cri-o/cri-o/actions/workflows/govulncheck.yml)
```
[the remark this answers is not in the document]
Comment (sohankunkerkar) [reply README.md:15]: But I don't see the badge when I use the display with the rich diff option

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -453,3 +453,32 @@ jobs:
...
+        run: |
+          report=`mktemp`
+          trap "rm $report" EXIT
+          `go env GOPATH`/bin/govulncheck -json -tags=test ./... > $report
+          modvulns=$(jq -Sr '.vulnerability.modules[]? | select(.path != "stdlib") | [.path, "affected package(s): \(.packages[].path)", "found version: \(.found_version)", "fixed version: \(.fixed_version)"]' < $report)
+          libvulns=$(jq -Sr '.vulnerability.modules[]? | select(.path == "stdlib") | [.path, "affected package(s): \(.packages[].path)", "found version: \(.found_version)", "fixed version: \(.fixed_version)"]' < $report)
+          echo "$modvulns"
+          echo "$libvulns"
+          # Exit nonzero iff there are any vulnerabilities in module dependencies
+          test -z "$modvulns"
```
Comment (saschagrunert) [inline .github/workflows/test.yml:482]: Let's add a dedicated script for this part.

File: Makefile [hunk 3]
```diff
@@ -61,6 +61,8 @@ MOCKGEN := ${BUILD_BIN_PATH}/mockgen
 MOCKGEN_VERSION := 1.6.0
 GOLANGCI_LINT := ${BUILD_BIN_PATH}/golangci-lint
 GOLANGCI_LINT_VERSION := v1.52.2
+GOSEC := ${BUILD_BIN_PATH}/gosec
+GOSEC_VERSION := 2.15.0
```
Comment (saschagrunert) [inline Makefile:65]: Let's add this to dependencies.yaml.

**Comment by haircommander** [comment]
/approve

LGTM, @cri-o/cri-o-maintainers PTAL

**Review by saschagrunert — APPROVED** [review]

**Pull request merged**